lib/render/Visualizer.cpp

```diff
@@ -491,28 +491,39 @@ RegionParams *Visualizer::getActiveRegionParams() const { return _paramsMgr->Get
 
 AnnotationParams *Visualizer::getActiveAnnotationParams() const { return _paramsMgr->GetAnnotationParams(_winName); }
 
+#include <vapor/STLUtils.h>
+
 int Visualizer::_captureImage(std::string path)
 {
     // Turn off the single capture flag
     _imageCaptureEnabled = false;
 
-    if (FileUtils::Extension(path) == "") path += ".png";
-
     ViewpointParams *vpParams = getActiveViewpointParams();
     int              width, height;
     //	vpParams->GetWindowSize(width, height);
     _framebuffer.GetSize(&width, &height);
 
-    bool geoTiffOutput = vpParams->GetProjectionType() == ViewpointParams::MapOrthographic && (FileUtils::Extension(path) == "tif" || FileUtils::Extension(path) == "tiff");
-
-    ImageWriter *  writer = nullptr;
     unsigned char *framebuffer = nullptr;
     int            writeReturn = -1;
 
     framebuffer = new unsigned char[3 * width * height];
     if (!_getPixelData(framebuffer))
         ;    // goto captureImageEnd;
 
+    if (STLUtils::BeginsWith(path, ":RAM:")) {
+        void *ptr;
+        sscanf(path.c_str(), ":RAM:%p", &ptr);
+
+        memcpy(ptr, framebuffer, sizeof(*framebuffer) * 3 * width * height);
+
+        delete[] framebuffer;
```

Comment:
Any reason to not use a C++ string and avoid the explicit memory management? There is already a goto commented out above that if were uncommented would result in a memory leak.

Reply:
I don't see how uncommenting the goto statement would result in a memory leak or how using a c++ string would help.

Reply:
I said "string" but I should have said C++ vector or array. I.e. a smart pointer that will clean up after itself when it goes out of scope.

Reply:
Done.